utils/modCommon.js

```diff
@@ -674,6 +674,44 @@ const modCommon = {
     }
   },
 
+  /**
+   * Give somebody a staff assigned role
+   * @param {Augur.GuildInteraction<"CommandSlash">} int
+   * @param {Boolean} give
+   * @param {Discord.GuildMember | null} recipient
+   * @returns {Promise<string>}
+   */
+  assignRole: async function(int, recipient, give = true) {
+    /** @param {Discord.GuildMember} member @param {string} id*/
+    const hasRole = (member, id) => member?.roles.cache.has(id);
+    try {
+      const pres = give ? "give" : "take";
+      const past = give ? "gave" : "took";
+      const input = int.options.getString("role", true);
+      if (!u.perms.calc(int.member, ["team", "mod", "mgr"])) return "*Nice try!* This command is for Team+ only";
+      if (!u.perms.calc(int.member, ["mod", "mgr"]) && ["adulting", "lady"].includes(input.toLowerCase())) return "This command is for Mod+ only";
+      if (!recipient) return `I couldn't find that user!`;
```

Comment:
recipient shouldn't be null. check for this in the main function and don't allow null into the modCommon parameter